homeassistant/components/tplink/const.py

```diff
@@ -19,3 +21,6 @@
 CONF_DEVICE_CONFIG: Final = "device_config"
 
 PLATFORMS: Final = [Platform.LIGHT, Platform.SENSOR, Platform.SWITCH]
+
+# List of modules that should be excluded from SWITCH platform creation.
+NO_SWITCH_DEVICE_MODULES = [Module.Light]
```

Comment:
This is to prevent creating a switch entity for lights. Once https://github.com/home-assistant/core/pull/117758 is merged this will no longer be required as switches will be dynamically created via Features that are not "Primary"

Reply:
Let's avoid adding this as it'd get removed anyway shortly.

Reply:
As agreed on discord we'll keep this in so have we have a single PR that does the migration to modules and a subsequent PR that introduces the features for switch. https://github.com/home-assistant/core/pull/117898 created to remove this in the next PR.